mage_ai/io/pinot.py

```diff
@@ -0,0 +1,163 @@
+import re
+from typing import IO, Any, Dict, List, Union
+
+from pandas import DataFrame, Series
+from pinotdb import connect
+from pinotdb.db import Connection
+
+from mage_ai.io.base import QUERY_ROW_LIMIT, ExportWritePolicy
+from mage_ai.io.config import BaseConfigLoader, ConfigKey
+from mage_ai.io.sql import BaseSQL
+
+WRITE_NOT_SUPPORTED_EXCEPTION = Exception('write operations are not supported.')
+
+
+class ConnectionWrapper(Connection):
```

Comment:
You don't necessarily need ConnectionWrapper if you don't override the cursor logic.